duckduckgo/macos-browser
# Add scan and optout support for child sites

## Asana task

https://app.asana.com/0/1204006570077678/1205508328452433/f

## Description

**Goal**: We will scan on the parent sites and child sites, but we will only opt out of the parent.

This will be the process:
- We do scans on both types of sites
- When we find matches on a child site, we create an opt-out operation but with the `preferredRunDate` set to `nil`, that’s because we do not run that operation, but we need it on the database to show it on the UI
- When an opt-out is finished on a parent site, we execute the new use case. The use case will take care of fetching all the child broker sites, and it will update their scan preferred run date to the confirm opt-out date set on its JSON file.

## Steps to test this PR

1. Do a search that has results both in the parent and child sites
2. Initialize the opt-out process
3. The only opt-out that should be run should be the parent. No child sites should run the opt-out
4. When the parent opt-out finishes, the scan preferred run date should change on the child sites
5. Then is business as usual. Scans will be run again on the parent and child, to either find new matches or confirm if a match was removed.
